src/unit/unit.service.ts

```diff
@@ -48,4 +60,53 @@ export class UnitService {
       where: { name: nameId, facId: facilityId },
     });
   }
+
+  private async getUnitDetails(unitId: number): Promise<UnitDTO[]> {
+    const sql = `
+        SELECT
+            unt.UNIT_ID as "id",
+            unt.UNITID as "unitid",
+            unt.NON_LOAD_BASED_IND as "nonLoadBasedIndicator",
+            unt.SOURCE_CATEGORY_CD as "sourceCategoryCd",
+            to_char(unt.COMM_OP_DATE, 'yyyy-mm-dd') AS "commOpDate",
+            to_char(unt.COMR_OP_DATE, 'yyyy-mm-dd') AS "comrOpDate",
+            (
+                SELECT ubt.UNIT_TYPE_CD
```

Comment:
I don't think that it is possible but you may want to put a MAX() around ubt.UNIT_TYPE_CD just in case non UNIT_BOILER_TYPE rows exists for a particular unit.

Reply:
Update made.